packages/core/src/analytics/dispatch.ts

```diff
@@ -57,3 +57,21 @@ export async function dispatch(
 
   return dispatched
 }
+
+/* Dispatch function, but swallow promise rejections and use event emitter instead */
+export const dispatchWithErrorHandling = async (
+  ...[event, queue, emitter, options]: Parameters<typeof dispatch>
+) => {
+  try {
+    const ctx = await dispatch(event, queue, emitter, options)
+    if (ctx.failedDelivery()) {
+      throw ctx
+    } else {
+      emitter.emit(event.type, ctx)
+      return ctx
+    }
+  } catch (ctx) {
+    emitter.emit('error', ctx)
```

Comment:
do we want to wrap all our errors in custom objects to give more metadata about their context -- some errors may not have actual context object.